File: Workshop 3/src/BlackJack/model/rules/BasicHitStrategy.java
/*
 * BasicHitStrategy.java
 * Date: 26 Oktober, 2014
 * Author: Tobias Ohlsson
 */
package BlackJack.model.rules;
import BlackJack.model.Player;

/*
 * Left untouched.
 */
class BasicHitStrategy implements IHitStrategy {
    
    private final int g_hitLimit = 17;

    @Override
    public boolean DoHit(Player a_dealer) {
        return a_dealer.CalcScore() < g_hitLimit;
    }
    
}

File: Workshop 3/Files/Peer Review/Materials - Eddy Proca/model/rules/BasicHitStrategy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BlackJack.model.rules
{
    class BasicHitStrategy : IHitStrategy
    {
        protected const int g_hitLimit = 17; // this may look like duplicate code but the limit may change depending on different styles of the game
        public bool DoHit(model.Player a_dealer)
        {
            return a_dealer.CalcScore() < g_hitLimit;
        }
    }
}
